src/error.rs

```diff
@@ -995,6 +997,29 @@ impl ToDiagnostic<FileId> for EvalError {
                     .with_labels(labels)
                     .with_notes(vec![String::from(INTERNAL_ERROR_MSG)])]
             }
+            EvalError::EqError { eq_pos, term: t } => {
+                let label = format!(
+                    "an argument has type {}, which cannot be compared for equality",
+                    t.term
+                        .type_of()
+                        .unwrap_or_else(|| String::from("<unevaluated>")),
+                );
+
+                let labels = match eq_pos {
+                    TermPos::Original(pos) | TermPos::Inherited(pos) if eq_pos != &t.pos => {
+                        vec![
+                            primary(pos).with_message(label),
+                            secondary_term(t, files)
+                                .with_message("problematic argument evaluated to this"),
+                        ]
+                    }
+                    _ => vec![primary_term(t, files).with_message(label)],
+                };
+
+                vec![Diagnostic::error()
+                    .with_message("equality error")
```

Comment:
The main error message is not very informative. Maybe "can't compare values for equality" or something like that?